### PR TITLE
Petr Dolejš - cleancode - NNPTP21-1

### PR DESCRIPTION
**Pull request by DOL7JS**
Renamed branch to dev

### CODE REVIEW

File: INPTPZ1/NewtonFractal.cs [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            }
+            if (complexNumber.Imaginary == 0)
+            {
+                complexNumber.Imaginary = 0.0001;
+            }
+        }
+        private float FindEquationSolutionWithNewtonsIteration(ref ComplexNumber complexNumber)
+        {
+            float iteration = 0;
+            for (int q = 0; q < 30; q++)
```
Comment (rodi0878) [inline INPTPZ1/NewtonFractal.cs]: why q?

File: INPTPZ1/NewtonFractal.cs [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+                complexNumber.Real = 0.0001;
+            }
+            if (complexNumber.Imaginary == 0)
+            {
+                complexNumber.Imaginary = 0.0001;
+            }
+        }
+        private float FindEquationSolutionWithNewtonsIteration(ref ComplexNumber complexNumber)
+        {
+            float iteration = 0;
```
Comment (rodi0878) [inline INPTPZ1/NewtonFractal.cs]: why float? i see only incrementation and not a single float/double operation

File: INPTPZ1/NewtonFractal.cs [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+                iteration++;
+            }
+
+            return iteration;
+        }
+
+        private int FindRootNumberSolution(ComplexNumber complexNumber)
+        {
+            var known = false;
+            var id = 0;
```
Comment (rodi0878) [inline INPTPZ1/NewtonFractal.cs]: index not id

File: INPTPZ1/NewtonFractal.cs [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                id = roots.Count;
+            }
+
+            return id;
+        }
+
+        private void ColorizeOnePixel(int i, int j, float it, int id)
+        {
+            var colorForPixel = COLORS[id % COLORS.Length];
+            colorForPixel = Color.FromArgb(colorForPixel.R, colorForPixel.G, colorForPixel.B);
```
Comment (rodi0878) [inline INPTPZ1/NewtonFractal.cs]: this line does nothing - remove

File: INPTPZ1/Program.cs [hunk 5]
```diff
@@ -25,309 +13,10 @@ class Program
...
-            //    {
-            //        Color c = bmp.GetPixel(j, i);
-            //        int nv = (int)Math.Floor(c.R * (255.0 / maxid));
-            //        bmp.SetPixel(j, i, Color.FromArgb(nv, nv, nv));
-            //    }
-            //}
-
-                    bmp.Save(output ?? "../../../out.png");
-            //Console.ReadKey();
+            NewtonFractal newtonFractal = new NewtonFractal(args);
```
Comment (rodi0878) [inline INPTPZ1/Program.cs:11]: i really dont like hiding all the work in the constructor. its not a good style; please create atleast one method which will do real work and call it. constructor is only for setting up of object and not for these calculations

**Review by rodi0878 — CHANGES_REQUESTED** [review]
There are few things that still need cleaning up. Please look at them and update your patch.

**Comment by DOL7JS** [comment]
Refactored

- Renamed variable in for cycle - q => i
- Refactored float to int
- Renamed id to index
- Removed useless line
- Created new method

**Comment by rodi0878** [comment]
accepted

**Pull request closed without merging**